src/core/dto/animatedModalDTO.ts

```diff
@@ -11,7 +14,23 @@ export interface AnimatedModalProps {
   closeSpaceVisibility?: number;
   hideKeyboardOnShow?: boolean;
   useKeyboardAvoidingView?: boolean;
-}
+};
+
+export type ControlledAnimatedModalProps = AnimatedModalBaseProps & {
```

Comment:
is this necessary?

Reply:
The separate `ControlledAnimatedModalProps` and `ForwardRefAnimatedModalProps` types, or the `controlled` prop? I think you're right that the `controlled` prop is unnecessary (getting rid of it now). The separate types, on the other hand, is so users (via TypeScript errors) do not try and implement both the `ref` and controlled states (they can either supply both `open` and `setOpen`, or `visible`, but not both)